xray/test/test_plotting.py

```diff
@@ -0,0 +1,242 @@
+        self.darray.plot_hist()
+        self.assertEqual('', plt.gca().get_title())
+
+    def test_title_uses_name(self):
+        self.darray.name = 'randompoints'
+        self.darray.plot_hist()
+        self.assertIn(self.darray.name, plt.gca().get_title())
+
+    def test_ylabel_is_count(self):
+        self.darray.plot_hist()
+        self.assertEqual('Count', plt.gca().get_ylabel())
+
+    def test_can_pass_in_kwargs(self):
+        nbins = 5
+        self.darray.plot_hist(bins=nbins)
+        self.assertEqual(nbins, len(plt.gca().patches))
+
+    def test_can_pass_in_axis(self):
+        self.pass_in_axis(self.darray.plot_hist)
+
+    def test_primitive_returned(self):
+        h = self.darray.plot_hist()
+        self.assertTrue(isinstance(h[-1][0], mpl.patches.Rectangle))
+
+    def test_plot_nans(self):
+        self.darray[0, 0, :] = np.nan
+        self.darray.plot_hist()
+
+
+class Common2dMixin:
+    """
+    Common tests for 2d plotting go here.
+
+    These tests assume that `self.plotfunc` exists and is defined in the
+    setUp. Should have the same name as the method.
+    """
+    def setUp(self):
+        self.darray = DataArray(np.random.randn(10, 15), dims=['y', 'x'])
+        self.plotmethod = getattr(self.darray, self.plotfunc.__name__)
+
+    def test_label_names(self):
+        self.plotmethod()
+        self.assertEqual('x', plt.gca().get_xlabel())
+        self.assertEqual('y', plt.gca().get_ylabel())
+
+    def test_1d_raises_valueerror(self):
+        with self.assertRaisesRegexp(ValueError, r'[Dd]im'):
+            self.plotfunc(self.darray[0, :])
+
+    def test_3d_raises_valueerror(self):
+        a = DataArray(np.random.randn(2, 3, 4))
+        with self.assertRaisesRegexp(ValueError, r'[Dd]im'):
+            self.plotfunc(a)
+
+    def test_nonnumeric_index_raises_typeerror(self):
+        a = DataArray(np.random.randn(3, 2),
+                      coords=[['a', 'b', 'c'], ['d', 'e']])
+        with self.assertRaisesRegexp(TypeError, r'[Ii]ndex'):
+            self.plotfunc(a)
+
+    def test_can_pass_in_axis(self):
+        self.pass_in_axis(self.plotmethod)
+
+    def test_xyincrease_false_changes_axes(self):
+        self.plotmethod(xincrease=False, yincrease=False)
+        xlim = plt.gca().get_xlim()
+        ylim = plt.gca().get_ylim()
+        diffs = xlim[0] - 14, xlim[1] - 0, ylim[0] - 9, ylim[1] - 0
+        self.assertTrue(all(abs(x) < 1 for x in diffs))
+
+    def test_xyincrease_true_changes_axes(self):
+        self.plotmethod(xincrease=True, yincrease=True)
+        xlim = plt.gca().get_xlim()
+        ylim = plt.gca().get_ylim()
+        diffs = xlim[0] - 0, xlim[1] - 14, ylim[0] - 0, ylim[1] - 9
+        self.assertTrue(all(abs(x) < 1 for x in diffs))
+
+    def test_plot_nans(self):
+        self.darray[0, 0] = np.nan
+        self.plotmethod()
+
+
+class TestContourf(Common2dMixin, PlotTestCase):
+
+    def setUp(self):
+        self.plotfunc = plot_contourf
```

Comment:
instead of overwriting `setUp`, you could just define plotfunc as a class variable, e.g., `plotfunc = plot_contourf`